arboretum/permissions/checks/test_org_permissions.py

```diff
@@ -0,0 +1,109 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Repository organization permissions check."""
+
+from compliance.check import ComplianceCheck
+from compliance.evidence import DAY, ReportEvidence, evidences
+from compliance.utils.data_parse import get_sha256_hash
+
+
+class OrgPermissionsCheck(ComplianceCheck):
+    """Monitor repository organization permissions."""
+
+    @property
+    def title(self):
+        """
+        Return the title of the checks.
+
+        :returns: the title of the checks
+        """
+        return 'Report on Repository Organization Permissions'
+
+    @classmethod
+    def setUpClass(cls):
+        """Initialize the check object with configuration settings."""
+        cls.config.add_evidences(
+            [
+                ReportEvidence(
+                    'org_permissions.md',
+                    'permissions',
+                    DAY,
+                    'Repository Permissions.'
+                )
+            ]
+        )
+        return cls
+
+    def test_org_permissions(self):
+        """Check the access to organization repositories."""
+        orgs = self.config.get('org.permissions.org_integrity.orgs')
+        for org in orgs:
+            host = org['url'].rsplit('/', 1)
+            service = 'gh'
+            if 'gitlab' in host:
+                service = 'gl'
+            elif 'bitbucket' in host:
+                service = 'bb'
+            url_hash = get_sha256_hash([org['url']], 10)
+            path = 'raw/permissions'
+            evidence_paths = {
+                'direct': f'{path}/{service}_direct_collaborators_'
+                + f'{url_hash}.json',
+                'outside': f'{path}/{service}_outside_collaborators_'
+                + f'{url_hash}.json',
+                'forks': f'{path}/{service}_forks_{url_hash}.json'
+            }
+            with evidences(self, evidence_paths) as ev:
+                self.add_failures(
+                    org['url'],
+                    self._check_collabs(
+                        ev['direct'].content_as_json,
+                        ev['outside'].content_as_json
+                    )
+                )
+                self.add_warnings(
+                    org['url'], self._check_forks(ev['forks'].content_as_json)
+                )
+
+    def _check_collabs(self, ev_direct, ev_outside):
+        repocollabs = []
+        for repo in ev_direct:
+            collabs = []
+            for c in ev_direct[repo]:
+                c['member'] = c not in ev_outside[repo]
+                collabs.append(c)
+            if not collabs:
+                continue
+            if collabs:
```

Comment:
When exactly would collabs not be collabs if it gets past the `continue`?  Seems like a useless `if` statement.

Reply:
Removed the useless statement